### PR TITLE
docs(config) document output.importFunctionName, externalsType and as…

### LINKED ISSUE #3748: Document webpack change: add support for async externals
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/10898

---

add `promise` external
add `import` external (uses import())

add `output.importFunctionName` option to change the `import()` function name

allow for inline external type when using arrays

fix some typings

fix namespace object behavior when using system external and accessing nested property
fix interop behavior for async dynamic modules

**What kind of change does this PR introduce?**

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
* add `promise` external (like `var` but awaits the result)
* add `import` external (uses import() to load a native ESM)
* `output.importFunctionName` allows to set `import()` function name when used for `import` external.

### PR DESCRIPTION
**Pull request by EugeneHlushko**
…ync externals changes

- document output.importFunctionName
- document externalsType
- add changes to mention async externals
- Closes #3748

### CODE REVIEW

File: src/content/configuration/output.md [hunk 1]
```diff
@@ -875,6 +874,12 @@ __system_context__.import('./other-file.js').then(m => {
 });
 ```
 
+### Asyncrhonous
```
Comment (chenxsan) [inline src/content/configuration/output.md:877]:
```suggestion
### Asynchronous
```

File: src/content/configuration/output.md [hunk 2]
```diff
@@ -886,6 +891,26 @@ MyLibrary(_entry_return_);
 The dependencies for your library will be defined by the [`externals`](/configuration/externals/) config.
 
 
+## `output.importFunctionName`
+
+`string = 'import'`
+
+The name of the native `import()` function used for importing the [`'import'` type externals](/configuration/output/#asyncrhonous).
```
Comment (chenxsan) [inline src/content/configuration/output.md:898]:
```suggestion
The name of the native `import()` function used for importing the [`'import'` type externals](/configuration/output/#asynchronous).
```

File: src/content/configuration/externals.md [hunk 3]
```diff
@@ -285,4 +290,43 @@ module.exports = {
...
+W> [Default type](/configuration/externals/#externalstype) will be used if you sepcify `externals` without a type e.g. `externals: { react: 'react' }` instead of `externals: { react: 'commonjs-module react' }`.
+
 For more information on how to use this configuration, please refer to the article on [how to author a library](/guides/author-libraries).
+
+
+## `externalsType`
+
+`string = 'var'`
+
+Specifies the default type of externals. `amd`, `root` and `system` externals __depend on the [`output.libraryTarget`](/configuration/output/#outputlibrarytarget) of the external__ being set to the same value e.g. you can consume `amd` externals within an `amd` library.
```
Comment (sokra) [inline src/content/configuration/externals.md:302]:
```suggestion
Specifies the default type of externals. `amd`, `root` and `system` externals __depend on the [`output.libraryTarget`](/configuration/output/#outputlibrarytarget)__ being set to the same value e.g. you can only consume `amd` externals within an `amd` library.
```

File: src/content/configuration/externals.md [hunk 4]
```diff
@@ -285,4 +290,43 @@ module.exports = {
...
+
+- `'var'`
+- `'module'`
+- `'assign'`
+- `'this'`
+- `'window'`
+- `'self'`
+- `'global'`
+- `'commonjs'`
+- `'commonjs2'`
```
Comment (sokra) [inline src/content/configuration/externals.md:314]:
```suggestion
```

deprecated in favor of `commonjs-module`

File: src/content/configuration/externals.md [hunk 5]
```diff
@@ -285,4 +290,43 @@ module.exports = {
...
+- `'commonjs-module'`
+- `'amd'`
+- `'amd-require'`
+- `'umd'`
+- `'umd2'`
+- `'jsonp'`
+- `'system'`
+- `'promise'` - same as `'var'` but awaits the result
+- `'import'` - uses `import()` to load a native EcmaScript module
```
Comment (sokra) [inline src/content/configuration/externals.md:323]:
```suggestion
- `'promise'` - same as `'var'` but awaits the result (async module, depends on `experiments.importAsync`)
- `'import'` - uses `import()` to load a native EcmaScript module (async module, depends on `experiments.importAsync`)
```

File: src/content/configuration/output.md [hunk 6]
```diff
@@ -886,6 +885,26 @@ MyLibrary(_entry_return_);
...
+
+__webpack.config.js__
+
+```javascript
+module.exports = {
+  //...
+  output: {
+    //...
+    libraryTarget: 'import',
+    importFunctionName: 'polyfillForImport'
```
Comment (sokra) [inline src/content/configuration/output.md:902]:
```suggestion
    importFunctionName: '__import__'
```

File: src/content/configuration/output.md [hunk 7]
```diff
@@ -886,6 +885,26 @@ MyLibrary(_entry_return_);
...
+The name of the native `import()` function. Can be used for polyfilling, e.g. with [`dynamic-import-polyfill`](https://github.com/GoogleChromeLabs/dynamic-import-polyfill).
+
+__webpack.config.js__
+
+```javascript
+module.exports = {
+  //...
+  output: {
+    //...
+    libraryTarget: 'import',
```
Comment (sokra) [inline src/content/configuration/output.md:901]:
```suggestion
```

**Pull request merged**